src/drivers/ihda.hpp

```diff
@@ -0,0 +1,95 @@
+#include <hw/audio_device.hpp>
```

Comment:
Are some of these parts potentially reusable? Having private structures used in private functions in this header they all then perhaps should be a separate class used as private here.

Reply:
No, the API is quite strange, and I dont even think the streams carrying the audio data will share anything. Maybe the audio stream rings are more generic.